routes/truckRoutes.js

```diff
@@ -1 +1,38 @@
-// put your Express router code in here
\ No newline at end of file
+var express = require('express');
+var trucks = require('../trucks');
+
+var router = express.Router();
+
+router.route('/')
+.get(function(request, response){
+  var truckList = trucks.getTrucks();
+  response.send(truckList);
+})
+.post(function(request, response){
+  var newTruck = trucks.addTruck(); 
```

Comment:
Hi James
I’ve re submitted my Module 5. A couple of things, I can’t get the add and remove to work correctly and when I lint my files I get can’t find express error. Any suggestions.
Thanks, Sean

From: jamdon 
Sent: Friday, May 20, 2016 12:55 AM
To: JSCRIPT300-Spring2016/Module_5 
Cc: seancava ; Author 
Subject: Re: [JSCRIPT300-Spring2016/Module_5] Mod 5 Submit (#9)

In routes/truckRoutes.js:

> \ No newline at end of file
> +var express = require('express');
> +var trucks = require('../trucks');
> +
> +var router = express.Router();
> +
> +router.route('/')
> +.get(function(request, response){
> +  var truckList = trucks.getTrucks();
> +  response.send(truckList);
> +})
> +.post(function(request, response){
> +  var newTruck = trucks.addTruck(); 
In your POST route, you need to get the request.body first, as this is what will contain the properties for the new truck. The addTruck method takes a truck object - since none was passed in, it will ultimately end up adding undefined.

—
You are receiving this because you authored the thread.
Reply to this email directly or view it on GitHub